src/ui/shared/round/view/roundView.tsx

```diff
@@ -305,13 +305,28 @@ function renderPlayerName(player: Player) {
   return 'Anonymous'
 }
 
+function getPlayerName(player: Player): string {
+  if (player.name != null || player.username != null || player.user) {
+    const name = player.name ?? player.username ?? player.user?.username
+    if (player.user?.title != null) {
+      return `${player.user.title} ${name}`
+    } else {
+      return name ?? 'Unknown'
+    }
+  }
+
+  if (player.ai != null) return playerApi.aiName({ ai: player.ai })
+
+  return 'Anonymous'
+}
+
 function renderAntagonistInfo(ctrl: OnlineRound, player: Player, material: Material, position: Position, isCrazy: boolean) {
   const user = player.user
   const playerName = renderPlayerName(player)
   const togglePopup = user ? () => ctrl.openUserPopup(position, user.id) : utils.noop
   const vConf = user ?
     helper.ontap(togglePopup, () => userInfos(user, player, playerApi.playerName(player))) :
-    helper.ontap(utils.noop, () => Plugins.LiToast.show({ text: playerName, duration: 'short' }))
+    helper.ontap(utils.noop, () => Plugins.LiToast.show({ text: getPlayerName(player), duration: 'short' }))
```

Comment:
is it fixing an issue here? (apart from typing the plugins as the PR title suggests)

Reply:
Possibly, I haven't tested out the UI interaction. `renderPlayerName` returns a `string` or a `Mithril.Vnode` which I don't think can be rendered as toast text, I haven't seen what happens if you try.